pilot/pkg/security/authn/v1alpha1/policy_applier.go

```diff
@@ -324,44 +349,20 @@ func (a v1alpha1PolicyApplier) InboundFilterChain(sdsUdsPath string, meta *model
 		ApplicationProtocols: util.ALPNInMesh,
 	}
 	tls := &auth.DownstreamTlsContext{
-		CommonTlsContext: &auth.CommonTlsContext{
-			// Note that in the PERMISSIVE mode, we match filter chain on "istio" ALPN,
-			// which is used to differentiate between service mesh and legacy traffic.
-			//
-			// Client sidecar outbound cluster's TLSContext.ALPN must include "istio".
-			//
-			// Server sidecar filter chain's FilterChainMatch.ApplicationProtocols must
-			// include "istio" for the secure traffic, but its TLSContext.ALPN must not
-			// include "istio", which would interfere with negotiation of the underlying
-			// protocol, e.g. HTTP/2.
-			AlpnProtocols: util.ALPNHttp,
-		},
-		RequireClientCertificate: protovalue.BoolTrue,
+		CommonTlsContext: &auth.CommonTlsContext{},
 	}
-	if sdsUdsPath == "" {
-		base := meta.SdsBase + constants.AuthCertsPath
-		tlsServerRootCert := model.GetOrDefault(meta.TLSServerRootCert, base+constants.RootCertFilename)
 
-		tls.CommonTlsContext.ValidationContextType = authn_model.ConstructValidationContext(tlsServerRootCert, []string{} /*subjectAltNames*/)
+	if sdsUdsPath == "" {
+		base := path.Join(meta.SdsBase, constants.AuthCertsPath)
+		tlsServerRootCert := model.GetOrDefault(meta.TLSServerRootCert, path.Join(base, constants.RootCertFilename))
 
-		tlsServerCertChain := model.GetOrDefault(meta.TLSServerCertChain, base+constants.CertChainFilename)
-		tlsServerKey := model.GetOrDefault(meta.TLSServerKey, base+constants.KeyFilename)
+		tlsServerCertChain := model.GetOrDefault(meta.TLSServerCertChain, path.Join(base, constants.CertChainFilename))
+		tlsServerKey := model.GetOrDefault(meta.TLSServerKey, path.Join(base, constants.KeyFilename))
 
-		tls.CommonTlsContext.TlsCertificates = []*auth.TlsCertificate{
-			{
-				CertificateChain: &core.DataSource{
-					Specifier: &core.DataSource_Filename{
-						Filename: tlsServerCertChain,
-					},
-				},
-				PrivateKey: &core.DataSource{
-					Specifier: &core.DataSource_Filename{
-						Filename: tlsServerKey,
-					},
-				},
-			},
-		}
+		setDownstreamTlsContext(tls, tlsServerRootCert, tlsServerCertChain, tlsServerKey, util.ALPNHttp, booltypes.BoolTrue, []string{})
 	} else {
+		tls.RequireClientCertificate = booltypes.BoolTrue
```

Comment:
Why does an else to `sdsUdsPath == ""` set **RequireClientCertificate**?

Reply:
it is set to true in both cases. for the non sds case the setDownstreamTlsContext function takes care of it.